### ISSUE TITLE
Why use idx_no_train ?

### ISSUE THREAD

**Issue by Chrisable** [BUPT-GAMMA/CPF#13]
idx_no_train contains test nodes and validation sets, but don't we usually use training nodes when calculating loss?

for example:
```
import torch.nn.functional as F

data = dataset[0]
optimizer = torch.optim.Adam(model.parameters(), lr=0.01)

for epoch in range(200):
    pred = model(data.x, data.edge_index)
    loss = F.cross_entropy(pred[data.train_mask], data.y[data.train_mask])

    # Backpropagation
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()
```

**Comment by J-cabin** [comment]
The reason is that the information of Labeled Node has been used as input, which is difficult to supervise.
